prody/dynamics/adaptive.py

```diff
@@ -1,140 +1,96 @@
+        atoms = None
 
-        self.coordsB = getCoords(structureB)
-        self.coordsB, _ = superpose(self.coordsB, self.coordsA)
-        self.coordsB_0 = self.coordsB
+    coordsB = getCoords(structureB)
+    
+    if title is None:
         if isinstance(structureB, Atomic):
-            self.titleB = structureB.getTitle()
+            title = structureB.getTitle()
+            atoms = structureB
         else:
-            self.titleB = 'structure B'
-
-        if self.coordsA.shape != self.coordsB.shape:
-            raise ValueError('structures should have the same number of atoms')
-
-        self.n_atoms = self.coordsA.shape[0]
-
-        rmsd = calcRMSD(self.coordsA, self.coordsB)
-        self.rmsds = [rmsd]
-        self.dList = []
-        self.numSteps = 0
-
-        self.n_modes = kwargs.get('n_modes', 20)
-        if not isinstance(self.n_modes, Integral):
-            raise TypeError('n_modes should be an integer')
-
-        self.cutoff = kwargs.get('cutoff', 15)
-        if not isinstance(self.cutoff, Number):
-            raise TypeError('cutoff should be a number')
-
-        self.anmListA = []
-        self.anmListB = []
-        self.numModesList = []
-        self.whichModesA = []
-        self.whichModesB = []
-
-        self.maxModes = kwargs.get('maxModes', None)
-        dof = 3*self.n_atoms-6
-        if self.maxModes is None:
-            self.maxModes = dof
+            title = 'Unknown'
+            atoms = None
 
-        if not isinstance(self.maxModes, (Integral, float)):
-            raise TypeError('maxModes should be an integer or float')
-        if self.maxModes < 1:
-            self.maxModes = int(self.maxModes * dof)
-        if self.maxModes > dof:
-            self.maxModes = dof
+    weightsA = structureA.getFlags("mapped") if isinstance(structureA, AtomMap) else 1.
+    weightsB = structureB.getFlags("mapped") if isinstance(structureB, AtomMap) else 1.
+    weights = weights * weightsA * weightsB
 
-        self.Fmin = kwargs.get('Fmin', None)
-        self.Fmin_max = kwargs.get('Fmin_max', 0.6)
+    if np.isscalar(weights):
+        weights = None
 
-        self.ensembleA = Ensemble(self.titleA)
-        self.ensembleA.addCoordset(self.coordsA)
+    coordsA, _ = superpose(coordsA, coordsB, weights)
+    rmsd = calcRMSD(coordsA, coordsB, weights)
+    LOGGER.info('Initialized Adaptive ANM with RMSD {:4.3f}\n'.format(rmsd))
 
-        self.ensembleB = Ensemble(self.titleB)
-        self.ensembleB.addCoordset(self.coordsB)
+    return coordsA, coordsB, title, atoms, weights, rmsd
 
-        self.ensemble = Ensemble('combined trajectory')
-        self.ensemble.setCoords(self.coordsA)
-
-        self.rmsd_diff_cutoff = kwargs.get('rmsd_diff_cutoff', 0.05)
-        self.target_rmsd = kwargs.get('target_rmsd', 1.0)
-
-        LOGGER.info(
-            'Initialised Adaptive ANM with RMSD {:4.3f}\n'.format(rmsd))
+def getTitle(structure, def_title='structure'):
+    if isinstance(structure, Atomic):
```

Comment:
could use has_attr but this is fine too

Reply:
Yes, but `isinstance` may be better in this case because it is more consistent with the function above.

Reply:
ok